File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/python3_solutions/2209-minimum-white-tiles-after-covering-with-carpets.py
class Solution:
    def minimumWhiteTiles(self, floor: str, numCarpets: int, carpetLen: int) -> int:
        n = len(floor)
        dp = [[0] * (numCarpets + 1) for _ in range(n + 1)]

        for i in range(1, n + 1):
            for j in range(numCarpets + 1):
                if j == 0:
                    dp[i][j] = dp[i - 1][j] + int(floor[i - 1])
                else:
                    dp[i][j] = min(dp[i - 1][j] + int(floor[i - 1]), dp[max(0, i - carpetLen)][j - 1])

        return dp[n][numCarpets]

File: leetcode/AIcoding/3.DeepSeek-v3/cpp_solutions/2209-minimum-white-tiles-after-covering-with-carpets.cpp
class Solution {
public:
    int minimumWhiteTiles(string floor, int numCarpets, int carpetLen) {
        int n = floor.size();
        vector<vector<int>> dp(numCarpets + 1, vector<int>(n + 1, 0));

        for (int i = 1; i <= n; ++i) {
            dp[0][i] = dp[0][i - 1] + (floor[i - 1] == '1' ? 1 : 0);
        }

        for (int k = 1; k <= numCarpets; ++k) {
            for (int i = 1; i <= n; ++i) {
                if (i <= carpetLen) {
                    dp[k][i] = 0;
                } else {
                    dp[k][i] = min(dp[k][i - 1] + (floor[i - 1] == '1' ? 1 : 0), dp[k - 1][i - carpetLen]);
                }
            }
        }

        return dp[numCarpets][n];
    }
};